exporter/terra/dcp_staging_client.py

```diff
@@ -0,0 +1,118 @@
+from exporter.metadata import MetadataResource, DataFile
+from exporter.graph.experiment_graph import LinkSet
+from typing import Iterable, IO, Dict, Any, Union
+
+from copy import deepcopy
+from io import StringIO
+
+from google.cloud import storage
+from mypy_boto3_s3 import S3Client
+import boto3
+import json
+
+
+Streamable = Union[bytes, StringIO, IO[Any]]
+
+
+class DcpStagingException(Exception):
+    pass
+
+
+class DcpStagingClient:
+
+    def __init__(self, s3_client: S3Client, gcs_client: storage.Client):
+        self.s3_client = s3_client
+        self.gcs_client = gcs_client
+
+    def write_metadatas(self, metadatas: Iterable[MetadataResource]):
+        for metadata in metadatas:
+            self.write_metadata(metadata)
+
+    def write_metadata(self, metadata: MetadataResource):
+        dest_object_key = f'metadata/{metadata.concrete_type()}/{metadata.uuid}_{metadata.dcp_version}.json'
+
+        if metadata.metadata_type == "file":
+            data_stream = DcpStagingClient.dict_to_json_stream(metadata.get_content())
+            self.write_to_staging_bucket(dest_object_key, data_stream)
+        else:
+            converted_file_metadata = DcpStagingClient.convert_file_metadata(metadata)
```

Comment:
in the old exporter logic, we're populating provenance fields, should it also be included here?

Reply:
Good point. I haven't seen anything explicit about it, but we probably will have to. Will confirm in the main design doc